### DISCUSSION TITLE
How to disable task timeout?

### DISCUSSION THREAD

**Discussion by dahaupt** [deployphp/deployer#2735] [Help needed]
The [documentation](https://deployer.org/docs/7.x/api#run) currently states: 

> timeout int|null — Sets the process timeout (max. runtime). The timeout in seconds (default: 300 sec; see {{default_timeout}}, null to disable).

Actually using `null` as parameter does not work because it will be [skipped](https://github.com/deployphp/deployer/blob/master/src/functions.php#L344). Is this intended or an issue?

**Our use case:** We run a `wget --spider` command via SSH in order to warm up frontend caches of our website. Depending on the project, this will take different amounts of time. We could alternatively just raise the timeout or start the `wget` process in the background on the remote server.

**Comment by antonmedv** [comment]
I think this is a bug, and 0 value should disable timeout.

**Reply by dahaupt** [reply]
The `setTimeout` method expects `null` in order to disable the timeout. Using `0` as value will also set the value to `null` within the `Process` class. So I will use that for now!
